### DISCUSSION TITLE
Why SASS is a devDependency?

### DISCUSSION THREAD

**Discussion by igotfr** [xania/view#6] [Q&A] [answered]
Why SASS is a peerDependency?

**Comment by igotfr** [comment]
- templates/gitmodule/src/time/time.module.scss
- templates/gitmodule/src/clock/clock.module.scss
- templates/default/src/clock/clock.module.scss
- templates/default/src/time/time.module.scss

**Comment by igotfr** [comment]
create a raw template that does not depend on sass, prettier and rxjs as they are optioned libs

**Reply by xania** [reply]
Hi, 
Point about prettier is a valid one
we need here an observable library and rxjs is most familiar
I have developed an alternative lib for rxjs called @xania/state with better performance than rxjs, Could be interesting to use that instead

**Reply by igotfr** [reply]
create a template without sass and prettier so

**Reply by xania** [reply]
But sass is needed though, isnt it? otherwise vite wont be able to compile scss files

**Reply by igotfr** [reply]
but what if i don't want to use sass? still need the devDependency?

**Reply by xania** [reply]
I see, it should be an option to select SASS. I will remove it for now and bring it back later in new template

**Reply by xania** [reply]
The templates are updates, would you mind let me know if all works for you. Make sure to use to the latest version of @xania/create-app v0.1.12

**Reply by igotfr** [reply]
$ npx create-xania txania2
npm ERR! code ENOVERSIONS
npm ERR! No versions available for create-xania

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/notecomm/.npm/_logs/2023-02-22T22_17_11_692Z-debug-0.log

**Reply by xania** [reply]
the package name is `@xania/create-app`

**Reply by xania** [reply]
npm init @xania/app txania2

**Reply by xania** [reply]
u should see this

![image](https://user-images.githubusercontent.com/34893682/220773387-6082ed90-0fb0-4f67-b8ac-b3370e025377.png)

**Reply by igotfr** [reply]
Sass and prettier are yet there

**Reply by xania** [reply]
What version are u using?
![image](https://user-images.githubusercontent.com/34893682/220775791-7d0d8bf3-b08e-41ac-9e46-149a3771262b.png)

**Reply by xania** [reply]
on windows I had to clear my npm cache with following

> npx rimraf %LocalAppData%\\npm-cache\\_npx

**Reply by igotfr** [reply]
I'm using 0.1.13

**Reply by xania** [reply]
weird

**Reply by xania** [reply]
This is what I see, 

![image](https://user-images.githubusercontent.com/34893682/220996840-b079e1c0-8b28-4043-9301-4f6b9a9af8a5.png)

Could you please provide more details?

**Reply by xania** [reply]
And this is how it looks like with _gitmodule_ template

![image](https://user-images.githubusercontent.com/34893682/220997577-325c026d-f4dd-4f8b-981e-a4550048d872.png)

**Reply by igotfr** [reply]
$ npm show @xania/create-app
```shell
@xania/create-app@0.1.13 | MIT | deps: 1 | versions: 11
Reactive UI Library

bin: create-app

dist
.tarball: https://registry.npmmirror.com/@xania/create-app/-/create-app-0.1.13.tgz
.shasum: c46bda0f6d08d88b61f947fcff6921835a5929e9
.integrity: sha512-b8NmHVIFKMmZMJTGgaxA6fLU/l6+4cf9Brx7yQ7X3v6yP0J9duO5/tFgFiu+DGoXQiSy86UdUOQHVj++ubpE8w==
.unpackedSize: 26.5 kB

dependencies:
create-create-app: ^7.3.0 

maintainers:
- xania <ibrahim.bensalah@gmail.com>

dist-tags:
latest: 0.1.13  

published 20 hours ago by xania <ibrahim.bensalah@gmail.com>
```

**Reply by igotfr** [reply]
I'm using Fedora, so I clean cache with $ npm cache clean --force

**Reply by xania** [reply]
do you have dependency to sass and prettier in your package.json of your newly created app with 0.1.13 version?

**Reply by igotfr** [reply]
what kind of details?

**Reply by igotfr** [reply]
> do you have dependency to sass and prettier in your package.json of your newly created app with 0.1.13 version?

yes

**Reply by igotfr** [reply]
recently, I had to change Javascript packages database server
```
$ npm config set registry https://registry.npm.taobao.org/
$ npm config get registry
```
https://github.com/npm/cli/issues/3359#issuecomment-1123207904

**Reply by xania** [reply]
It is removed though from the templates, we can verify here:
https://unpkg.com/browse/@xania/create-app@0.1.13/templates/gitmodule/package.json 
and here
https://unpkg.com/browse/@xania/create-app@0.1.13/templates/default/package.json

**Reply by xania** [reply]
I havent seen this registry before, could you verify it's not an issue with that registry?

**Reply by xania** [reply]
I have to admin, using npm init to install the app is the worst, there seems to always be some issue with it.

That's why I was exploring another option using enquirer, seems like the most flexible solution but the work needed is huge. It gonna take me another month or two before I can start working on it.

**Reply by igotfr** [reply]
> do you have dependency to sass and prettier in your package.json of your newly created app with 0.1.13 version?

I newly created app with 0.1.13

**Reply by xania** [reply]
> > do you have dependency to sass and prettier in your package.json of your newly created app with 0.1.13 version?
> 
> I newly created app with 0.1.13

and? prettier and sass still there?

**Reply by igotfr** [reply]
yes

**Reply by igotfr** [reply]
@xania I tried in on other computer on same network and it works, do you know clean cache, I tried $ npm cache clean --force but it didn't work

**Reply by xania** [reply]
> @xania I tried in on other computer on same network and it works, do you know clean cache, I tried $ npm cache clean --force but it didn't work

npm cache clean --force doest work for me either, this is what I use on windows machine

> npx rimraf %LocalAppData%\npm-cache\_npx

**Reply by igotfr** [reply]
you could use $ npx degit to generate templates, like Solid does, $ npm init is buggy

**Reply by xania** [reply]
I was thinking of using a combination of both, npm init will become an empty shell which dynamically load available templates, using degit. That would solve the issue of the npm init cache because the templates are external.

**Reply by xania** [reply]
once the app is generated then you are basically stuck with that choice, enquirer would be a solution to be able to update the template after creating it, like angular cli. but as I mentioned, that's a lot more work

**Reply by igotfr** [reply]
I cleared the cache with:
`$ npx rimraf ~/.npm/_npx/`

**Comment by xania** [comment, marked as answer]
> Why SASS is a peerDependency?

Do you mean devDepwndency?

**Reply by igotfr** [reply]
yes
